### PR TITLE
chore(tracer): deprecate tracer.write method

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The write method is no longer used by the library and it's not a
use-case that we'd like to support.

If a user desires to write spans to a writer then it can be done
manually by using a writer reference.

### CODE REVIEW

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.59

File: releasenotes/notes/tracer-write-5d46dc5c4e870b9b.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+deprecations:
+  - |
+    Tracer.write() is deprecated. Directly writing spans to the writer is not supported.
```
Comment (brettlangdon) [inline releasenotes/notes/tracer-write-5d46dc5c4e870b9b.yaml:4]:
```suggestion
    ``Tracer.write()`` is deprecated. Directly writing spans to the writer is not supported.
```

**Review by brettlangdon — APPROVED** [review]

**Review by brettlangdon — DISMISSED** [review]

**Review by majorgreys — DISMISSED** [review]

**Review by majorgreys — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
```
  Traceback (most recent call last):
    File "/project/tests/smoke_test.py", line 3, in <module>
      import ddtrace.appsec._ddwaf
    File "/tmp/tmp.HhvL8XbIEH/venv/lib/python2.7/site-packages/ddtrace/__init__.py", line 7, in <module>
      from .tracer import Tracer  # noqa: E402
    File "/tmp/tmp.HhvL8XbIEH/venv/lib/python2.7/site-packages/ddtrace/tracer.py", line 90, in <module>
      class Tracer(object):
    File "/tmp/tmp.HhvL8XbIEH/venv/lib/python2.7/site-packages/ddtrace/tracer.py", line 815, in Tracer
      removal_version="1.0.0",
  TypeError: 'NoneType' object is not callable
```

**Comment by Kyle-Verhoog** [comment]
🤔 🤔 🤔

**Review by brettlangdon — APPROVED** [review]

**Comment by mabdinur** [comment]
@Mergify backport 0.59

**Review by mabdinur — APPROVED** [review]

**Pull request merged**